content/client-lib-development-guide/features.textile

```diff
@@ -1628,20 +1636,27 @@ class Push:
 class PushAdmin:
   deviceRegistrations: PushDeviceRegistrations
   channelSubscriptions: PushChannelSubscriptions
+  publish(recipient: JsonObject, data: JsonObject) => io
 
 class JsonObject:
   // Platform-dependant, typically a Dict-like object
 
 class PushDeviceRegistrations:
+  get(device: DeviceDetails) => io DeviceDetails
+  get(deviceId: String?) => io DeviceDetails
+  list(params: Dict<String, String>?) => io PaginatedResult<DeviceDetails>
   save(DeviceDetails) => io
-  get(params: Dict<String, String>?) => io PaginatedResult<DeviceDetails>
-  remove(params: Dict<String, String>?) => io
+  remove(device: DeviceDetails) => io
+  remove(deviceId: String?) => io
+  removeWhere(params: Dict<String, String>?) => io
 
 class PushChannelSubscriptions:
-  save(PushChannelSubscription) => io
-  listChannels() => io PaginatedResult<String>
-  get(params: Dict<String, String>?) => io PaginatedResult<PushChannelSubscription>
-  remove(params: Dict<String, String>?) => io
+  get(pushChannelSubscription: PushChannelSubscription) => io PushChannelSubscription
```

Comment:
We don't really need to give a name to this parameter.

Returns `PushChannelSubscription?`.